### PR TITLE
CI/BUG/TST: Add Kudu to CI and fix kudu table discovery bug

### LINKED ISSUE #1509: ValueError when creating a table expression for a Kudu-backed Impala table
I believe the issue I'm experiencing is that `DESCRIBE` returns more than just three columns for Kudu-backed tables (see "Kudu Considerations" [on this page](https://www.cloudera.com/documentation/enterprise/5-11-x/topics/impala_describe.html)). The issue seems to be

https://github.com/ibis-project/ibis/blob/3276d3713c3497021e547e10102f917f7762414d/ibis/impala/client.py#L1060 

which assumes there are only three columns. 

Example traceback:

    In [5]: client.table('kudu_backed_table')
    ---------------------------------------------------------------------------
    ValueError                                Traceback (most recent call last)
    <ipython-input-5-d0d66416fb5f> in <module>()
    ----> 1 client.table('kudu_backed_table')
    
    /data/apps/Anaconda3-5.0.1/envs/ibis/lib/python3.6/site-packages/ibis/client.py in table(self, name, database)
        117         """
        118         qualified_name = self._fully_qualified_name(name, database)
    --> 119         schema = self._get_table_schema(qualified_name)
        120         node = ops.DatabaseTable(qualified_name, schema, self)
        121         return self._table_expr_klass(node)
    
    /data/apps/Anaconda3-5.0.1/envs/ibis/lib/python3.6/site-packages/ibis/impala/client.py in _get_table_schema(self, tname)
       1186
       1187     def _get_table_schema(self, tname):
    -> 1188         return self.get_schema(tname)
       1189
       1190     def _get_schema_using_query(self, query):
    
    /data/apps/Anaconda3-5.0.1/envs/ibis/lib/python3.6/site-packages/ibis/impala/client.py in get_schema(self, table_name, database)
        716         tuples = self.con.fetchall(query)
        717
    --> 718         names, types, comments = zip(*tuples)
        719
        720         ibis_types = []
    
    ValueError: too many values to unpack (expected 3)

### LINKED ISSUE #1547: Move kudu testing image underneath ibisproject on dockerhub

### PR DESCRIPTION
**Pull request by cpcloud**
Closes #1509 
Closes #1547

### CODE REVIEW

**Comment by cpcloud** [comment]
I'm going to remove the installation of the kudu client for now, since that seems to be failing to find a particular symbol (I'm probably installing shared libraries that are incompatible with the version of `kudu-python` that I'm trying to install) and will make a follow up issue to track that problem.

File: ci/docker-compose.yml [hunk 1]
```diff
@@ -45,11 +46,40 @@ services:
 
   clickhouse:
     image: yandex/clickhouse-server:1.1.54388
-
     ports:
       - 8123:8123
       - 9000:9000
 
+  kudu-master:
+    image: cpcloud86/kudu:latest
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ci/docker-compose.yml]: Yep

File: ci/docker-compose.yml [hunk 2]
```diff
@@ -45,11 +46,40 @@ services:
...
+    cap_add:
+      - SYS_TIME
+    ports:
+      - 7050:7050
+      - 8050:8050
+    environment:
+      KUDU_MASTER: "false"
+    volumes:
+      - /etc/timezone:/etc/timezone:ro
+      - /etc/localtime:/etc/localtime:ro
```
Comment (kszucs) [inline ci/docker-compose.yml]: I'm afraid this docker-compose will run dozens of services in the near future :D

**Comment by kszucs** [comment]
How about asking the Kudu developers to create a conda forge feedstock?

File: ci/docker-compose.yml (hunk 2 above)
Comment (cpcloud) [reply ci/docker-compose.yml]: I was thinking we should break up the impala image into `hadoop`, `hive`, and `impala`. Is that completely insane?

**Comment by cpcloud** [comment]
Merging on green.

**Pull request closed without merging**